presto-spi/src/main/java/com/facebook/presto/spi/block/VariableWidthBlockBuilder.java

```diff
@@ -275,6 +275,13 @@ public void reset(BlockBuilderStatus blockBuilderStatus)
         updateArraysDataSize();
     }
 
+    @Override
+    public BlockBuilder newBlockBuilderLike(BlockBuilderStatus blockBuilderStatus)
+    {
+        int expectedBytesPerEntry = positions == 0 ? positions : (getOffset(positions) - getOffset(0)) / positions;
```

Comment:
We cannot add a new constructor like `public VariableWidthBlockBuilder(BlockBuilderStatus blockBuilderStatus, int expectedEntries, int expectedSizeInBytes)` as you recommended, because the existing constructor has the same method signature.

So I decided to use `0` as the `expectedBytesPerEntry` when `positions` is `0` as this is exactly how `reset()` behaves right now.